simplq/src/components/pages/Status/NotificationContainer.jsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import Switch from '@material-ui/core/Switch';
+import InfoOutlinedIcon from '@material-ui/icons/InfoOutlined';
+import { setNotificationPermission } from '../../../store/appSlice';
+import styles from '../../../styles/statusPage.module.scss';
+
+export default () => {
+  const dispatch = useDispatch();
+  const notificationPermission = useSelector((state) => state.appReducer.notificationPermission);
+
+  const hasPromiseBasedNotificationSupport = () => {
+    try {
+      Notification.requestPermission().then();
+    } catch (e) {
+      return false;
+    }
+
+    return true;
+  };
+
+  const askNotificationPermission = () => {
+    // function to actually ask the permissions
+    const handlePermission = (permission) => {
+      // Some older versions of chrome doesn't store the permission in the Notification object. In that case we store it manually.
+      if (!('permission' in Notification)) {
+        Notification.permission = permission;
+      }
+
+      dispatch(setNotificationPermission(permission));
+    };
+
+    // Let's check if the browser supports notifications
+    if (!('Notification' in window)) {
+      // eslint-disable-next-line no-console
+      console.log('This browser does not support notifications.');
+    } else if (hasPromiseBasedNotificationSupport()) {
+      Notification.requestPermission().then((permission) => {
+        handlePermission(permission);
+      });
+    } else {
+      Notification.requestPermission((permission) => {
+        handlePermission(permission);
+      });
+    }
+  };
+
+  if (notificationPermission === 'default') {
+    return (
+      <div className={styles['notification-container']}>
+        <Switch
+          classes={{ thumb: 'switch-thumb' }}
+          onChange={askNotificationPermission}
+          size="small"
+        />
+        <span>Enable notification</span>
+        <InfoOutlinedIcon classes={{ root: 'info-icon' }} />
+      </div>
+    );
+  }
+
+  if (notificationPermission === 'denied') {
+    return <>*If required, please enable notifications in the page settings</>;
+  }
+  return null;
```

Comment:
Could you try modifying the container so that the toggle is always present instead of this message? On toggling on, the adk function has to be called and if denied, it toggles off.

Reply:
Done. By the way, if the user denies the modal dialoge (sets notifications to denied) then clicking on the switch won't do anything, as `Notification.requestPermission()` does not toggle the modal again (I guess this was done in order to protect the user from spamming). So maybe there should be something like an explanation why this behaves the way it does and tell the user how to activate the notifications.

Reply:
I'm not completely sure what you mean. Could you push your latest changes here first? I'll take a look and maybe that'll make things clear.

Reply:
Right, I forgot about this. Let's leave it like this for now. In a later PR, we should maybe grey it out and then add a tooltip. Can you raise a issue to track this?

Reply:
Also, do try to fix the merge conflicts before pushing.

Reply:
Ok, seems like the conflicts are fixed.

Reply:
The toggle works fine. I prefer letting the user enable notifications later on toggling it on even if they denied once, but I'm not sure how easy it is to implement. We'll decide and try that out later someday.

Reply:
I would also prefer it that way, but on quick glance over the documentation, I did not find an obvious non hacky way to do it.